### ISSUE TITLE
[Bug]: `react-hooks/exhaustive-deps` false positive for object literal shorthand dependencies

### ISSUE THREAD

**Issue by elecmonkey** [web-infra-dev/rslint#961]
### System Info

  System:
    OS: macOS 26.4.1
    CPU: (18) arm64 Apple M5 Pro
    Memory: 3.46 GB / 48.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 24.15.0 - /Users/bytedance/.nvm/versions/node/v24.15.0/bin/node
    npm: 11.12.1 - /Users/bytedance/.nvm/versions/node/v24.15.0/bin/npm
    pnpm: 11.1.2 - /Users/bytedance/.nvm/versions/node/v24.15.0/bin/pnpm
    bun: 1.3.14 - /Users/bytedance/.bun/bin/bun
  Browsers:
    Chrome: 147.0.7727.56
    Safari: 26.4

### Details

`react-hooks/exhaustive-deps` false positive for object literal shorthand dependencies

### Reproduce link

https://github.com/elecmonkey/rslint-exhaustive-deps-bug-report

### Reproduce Steps

clone my repro repo and run: 

```bash
pnpm lint:eslint
pnpm lint:rslint
```

minimal reproduction component:
```tsx
import { useMemo } from 'react';

const App = () => {
  const isAdmin = true;

  const value = useMemo(() => ({ isAdmin }), [isAdmin]);

  return <pre>{String(value.isAdmin)}</pre>;
};

export default App;
```
